lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1005,7 +1005,7 @@ bool TypeChecker::typeCheckCatchPattern(CatchStmt *S, DeclContext *DC) {
 
 static bool isDiscardableType(Type type) {
   return (type->is<ErrorType>() ||
-          type->isNever() ||
+          type->isUninhabited() ||
```

Comment:
Actually I don't understand why, for example, we don't warn about unused `Optional<()>` results. Can you give me a refresher?